### PR TITLE
use nim provided language model

### PR DESCRIPTION
**Pull request by osmaczko**
Use nim provided language model in qml.
closes: https://github.com/status-im/status-desktop/issues/6416
StatusQ: https://github.com/status-im/StatusQ/pull/797

### Affected areas

changing language

### CODE REVIEW

**Review by stefandunca — APPROVED** [review]

**Review by caybro — CHANGES_REQUESTED** [review]
Just a minor naming problem, otherwise looks good

**Review by noeliaSD — CHANGES_REQUESTED** [review]
Testing it, and based on current master behaviour, I have 2 questions:

1. How the searcher works? I was expecting it takes into account the complete item name but it just only takes into account if it exactly matches the item name from the starting point. On the current master you can write "brazil" or "taiwan" and it searches the correct item. Here a video of this branch behaviour:

https://user-images.githubusercontent.com/97019400/180944997-de71cd1d-bac9-491c-9614-76f95904f8de.mov

2. Why can I only see some of this `locale_table` options? Is it the expected behaviour? On master we have more visible languages to select. Here a video of the ones I can see.

https://user-images.githubusercontent.com/97019400/180945021-fb22e600-a05c-4c83-a984-77bd9382ea4d.mov

**Comment by osmaczko** [comment]
@noeliaSD 

AD1) How the search wors is defined in `StatusListPicker`. When I was refactoring `StatusListPicker` I changed `searchText.includes` to `searchText.startsWith` by mistake. Now it should behave as before.

AD2) `locale_table` is just a table with a description of all possible locales. Which languages the app really uses is defined by *.ts (*.qm to be more precise) files and these [come from the Lokalise platform](https://github.com/status-im/status-desktop/pull/6568).

File: ui/app/AppLayouts/Profile/stores/LanguageStore.qml [hunk 1]
```diff
@@ -5,50 +5,11 @@ QtObject {
...
-                root.languageModel.get(i).selected = false
-            }
-        }
-
-        // Set default:
-        if(!isSelected)
-            root.languageModel.get(0).selected = true
-    }
+    readonly property var languageModel: languageModule ? languageModule.model : null
+    readonly property string currentLocale: languageModule ? languageModule.currentLocale : null
```
[the remark this answers is not in the document]
Comment (osmaczko) [reply ui/app/AppLayouts/Profile/stores/LanguageStore.qml]: Yup, that's right. Changed "locale" to "translation": https://github.com/status-im/status-desktop/pull/6590/commits/e0e5e909f54d42a8e06b388882d10ec027265643. 

Still, the "locale" term is misused in some places in qml code, but I would like to keep it out of scope for this PR.

[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Profile/stores/LanguageStore.qml]: Definitely, here's a good start :)

**Review by caybro — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/views/LeftTabView.qml [hunk 2]
```diff
@@ -58,7 +58,7 @@ Rectangle {
                 id: walletAmountValue
                 color: Style.current.textColor
                 text: {
-                    Utils.toLocaleString(parseFloat(RootStore.totalCurrencyBalance).toFixed(2), localAppSettings.locale, {"currency": true}) + " " + RootStore.currentCurrency.toUpperCase()
+                    Utils.toLocaleString(parseFloat(RootStore.totalCurrencyBalance).toFixed(2), localAppSettings.language, {"currency": true}) + " " + RootStore.currentCurrency.toUpperCase()
```
[the remark this answers is not in the document]
Comment (osmaczko) [reply ui/app/AppLayouts/Wallet/views/LeftTabView.qml:61]: > Something to fix globally I guess

Yes, definitely. Do we have a task for that already created?

> Rationale: https://hackmd.io/@status-desktop/B1eOaf-nd/%2FjNzZnH8ZRiynrrXJ--XpmQ#Currency-values

Great summary :+1:

[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Wallet/views/LeftTabView.qml:61]: Nope, I'll create one

**Comment by noeliaSD** [comment]
> @noeliaSD
> 
> AD1) How the search wors is defined in `StatusListPicker`. When I was refactoring `StatusListPicker` I changed `searchText.includes` to `searchText.startsWith` by mistake. Now it should behave as before.
> 
> AD2) `locale_table` is just a table with a description of all possible locales. Which languages the app really uses is defined by _.ts (_.qm to be more precise) files and these [come from the Lokalise platform](https://github.com/status-im/status-desktop/pull/6568).

Thanks for the clarification!

**Review by noeliaSD — APPROVED** [review]
Tested! LGTM!

**Pull request merged**